backend/routes/songs.js

```diff
@@ -120,6 +120,59 @@ app.put("/likes/:id", (req, res, next) => {
     });
 });
 
+
+
+
+
+app.get("/search", (req, res, next) => {
+    console.log(`user seesion: ${req.session.userId}`);
+    const pageSize = +req.query.pageSize;
+    const currPage = +req.query.page;
+    const songName = req.query.songName;
+    const artistName = req.query.artistName;
+    const genreName = req.query.genreName;
+
+    let fetchedSongs;
+    let query = {};
+
+    if(songName!=='') query["name"] = songName;
+    if(artistName!=='') query["artists.display_name"] = artistName;
+    if(genreName!=='') query["genre"] = genreName;
+
+
+    const songQuery = Song.find(query);
+
+    Song.find(query).then(results=>{
+      totalNumFiltered = results.length;
+    });
+
+
+    if (pageSize && currPage){
+        songQuery.skip(pageSize * (currPage - 1)).limit(pageSize);
```

Comment:
can't use songQuery if you've used Song.find

Reply:
that what alex did

Reply:
are you sure you cant?

Reply:
You have to use the songQuery in find otherwise the then callback might be called before the results are ready. (line 145)